packages/embark-mocha-tests/src/lib/index.js

```diff
@@ -0,0 +1,151 @@
+const assert = require('assert').strict;
+const async = require('async');
+const Mocha = require('mocha');
+const Web3 = require('web3');
+
+const Reporter = require('./reporter');
+
+const JAVASCRIPT_TEST_MATCH = /^.+\.js$/i;
+const TEST_TIMEOUT = 15000; // 15 seconds in milliseconds
+
+class MochaTestRunner {
+  constructor(embark, options) {
+    this.embark = embark;
+
+    this.files = [];
+
+    const { events } = embark;
+    events.request('tests:runner:register',
+      'JavaScript (Mocha)',
+      this.match.bind(this),
+      this.addFile.bind(this),
+      this.run.bind(this)
+    );
+  }
+
+  addFile(path) {
+    if (!this.match(path)) {
+      throw new Error(`invalid JavaScript test path: ${path}`);
+    }
+
+    this.files.push(path);
+  }
+
+  match(path) {
+    return JAVASCRIPT_TEST_MATCH.test(path);
+  }
+
+  run(options, cb) {
+    const {events} = this.embark;
+
+    const Module = require("module");
+    const originalRequire = require("module").prototype.require;
+
+    let accounts = [];
+    let compiledContracts = {};
+    let web3;
+
+    const config = (cfg, acctCb) => {
+      global.before((done) => {
+        async.waterfall([
+          (next) => {
+            events.request("contracts:build", cfg, compiledContracts, next);
+          },
+          (contractsList, contractDeps, next) => {
+            events.request("deployment:contracts:deploy", contractsList, contractDeps, next);
+          },
+          (_foo, next) => {
+            events.request("contracts:list", next);
+          },
+          (contracts, next) => {
+            for(const contract of contracts) {
+              const instance = compiledContracts[contract.className];
+              const contractObj = new web3.eth.Contract(instance.abiDefinition, contract.deployedAddress);
+              Object.setPrototypeOf(compiledContracts[contract.className], contractObj);
+            }
+
+            next();
+          }
+        ], (_err) => {
```

Comment:
Could it be:
```
acctCb(_err, accounts)
```
to pass the error on in case there is an error. Or possibly a log to inform the user:
```
if(_err) {
  this.logger.error(`Error running tests: ${_err.message || err}`);
}
acctCb(null, accounts);
```